src/accelerate/test_utils/scripts/test_sync.py

```diff
@@ -202,11 +202,13 @@ def test_gradient_accumulation_with_opt_and_scheduler(split_batches=False, dispa
         ddp_model.train()
         step_model(model, input, target, accelerator, False)
         opt.step()
-        if split_batches:
-            sched.step()
-        else:
-            for _ in range(accelerator.num_processes):
+
+        if ((iteration + 1) % 2 == 0) or ((iteration + 1) == len(dataloader)):
+            if split_batches:
                 sched.step()
+            else:
+                for _ in range(accelerator.num_processes):
+                    sched.step()
```

Comment:
Could you explain he logic here?

Reply:
Yes, basically I added if-statement to check  that optimizer step was made, as in line 224: `did_step = (((iteration + 1) % 2) == 0) or ((iteration + 1) == len(dataloader)) `. Because AcceleratedScheduler now steps every `gradient_accumulation_steps` steps, I added same logic to the ground truth scheduler. May be I should have moved `opt.step()` under this if statement too.